tiledb/sm/query/query.cc

```diff
@@ -784,14 +784,18 @@ Layout Query::layout() const {
   return layout_;
 }
 
-const QueryCondition& Query::condition() const {
+QueryCondition Query::condition() const {
```

Comment:
Why are we changing the return from a reference? Won't this now create a new object and potentially cause performance problems if called in a high performance codepath? For now it would be ok because this is only used in the serialization code paths... But who knows how it will be used moving forward. This would actually return a reference to the optional and have the serialization path deal with has_value. Also, I'm not a big fan of return QueryCondition(); because this will prevent removing the default constructor for the query condition which we should do for C.41 compliance.

Reply:
Had to undo this to remember why, and it was because, well, what do we return if our `std::optional<QueryCondition>` is empty? I'll change the return value to `const std::optional<QueryConditonal>&` and see how far that spiders out instead.